test/GPv2Settlement.test.ts

```diff
@@ -230,55 +230,99 @@ describe("GPv2Settlement", () => {
           sellPrice,
           executedBuyAmount,
           buyPrice,
-        } = await processTradeVariant(OrderKind.SELL, false);
+        } = await computeTradeExecutionVariant(OrderKind.SELL, false);
 
         expect(executedSellAmount).to.deep.equal(sellAmount);
         expect(executedBuyAmount).to.deep.equal(
-          sellAmount.mul(buyPrice).div(sellPrice),
+          sellAmount.mul(sellPrice).div(buyPrice),
         );
       });
 
+      it("should respect the limit price for fill-or-kill sell orders", async () => {
+        const { executedBuyAmount } = await computeTradeExecutionVariant(
+          OrderKind.SELL,
+          false,
+        );
+
+        expect(executedBuyAmount.gt(buyAmount)).to.be.true;
+      });
+
       it("should compute amounts for fill-or-kill buy orders", async () => {
         const {
           executedSellAmount,
           sellPrice,
           executedBuyAmount,
           buyPrice,
-        } = await processTradeVariant(OrderKind.BUY, false);
+        } = await computeTradeExecutionVariant(OrderKind.BUY, false);
 
         expect(executedSellAmount).to.deep.equal(
-          buyAmount.mul(sellPrice).div(buyPrice),
+          buyAmount.mul(buyPrice).div(sellPrice),
         );
         expect(executedBuyAmount).to.deep.equal(buyAmount);
       });
 
+      it("should respect the limit price for fill-or-kill buy orders", async () => {
+        const { executedSellAmount } = await computeTradeExecutionVariant(
+          OrderKind.BUY,
+          false,
+        );
+
+        expect(executedSellAmount.lt(sellAmount)).to.be.true;
+      });
+
       it("should compute amounts for partially fillable sell orders", async () => {
         const {
           executedSellAmount,
           sellPrice,
           executedBuyAmount,
           buyPrice,
-        } = await processTradeVariant(OrderKind.SELL, true);
+        } = await computeTradeExecutionVariant(OrderKind.SELL, true);
 
         expect(executedSellAmount).to.deep.equal(executedAmount);
         expect(executedBuyAmount).to.deep.equal(
-          executedAmount.mul(buyPrice).div(sellPrice),
+          executedAmount.mul(sellPrice).div(buyPrice),
         );
       });
 
+      it("should respect the limit price for partially fillable sell orders", async () => {
+        const {
+          executedSellAmount,
+          executedBuyAmount,
+        } = await computeTradeExecutionVariant(OrderKind.SELL, true);
+
+        expect(
+          executedBuyAmount
+            .mul(sellAmount)
+            .gt(executedSellAmount.mul(buyAmount)),
```

Comment:
If "gt" means "greater than", shouldn't it be "greater or equal" ?

Reply:
If I got it right, there is only one order that is matched at a fixed price that is different from the order price. So if the order is matched, then the price is expected to not be equal. In all generality it could be equal, but for the sake of the test I like it more without the equal since in that case you are not really testing the expected behavior (if it's to the left or to the right of the price).